packages/modal/src/Modal.js

```diff
@@ -1,6 +1,72 @@
-import { Modal } from "hig-react";
-import "./modal.scss";
+import React, { Component } from "react";
+import PropTypes from "prop-types";
 
-Modal.displayName = "Modal";
+import ModalPresenter from "./presenters/ModalPresenter";
+import ModalBehavior from "./behaviors/ModalBehavior";
 
-export default Modal;
+export default class Modal extends Component {
+  static propTypes = {
+    /**
+     * Supports adding any dom content to the body of the modal
+     */
+    children: PropTypes.node,
+    /**
+     * Supports adding any dom content to the header of the modal
+     */
+    headerChildren: PropTypes.node,
+    /**
+     * Triggers when you click the close button
+     */
+    onCloseClick: PropTypes.func,
```

Comment:
I wonder if we should expose more than `onCloseClick` and `onOverlayClick` here.  This is maintaining the vanilla API

Reply:
If this component had uncontrolled functionality, more handlers like `onClose` would be nice. In this case, these are fine for now.

Reply:
🆗